include/oneapi/tbb/detail/_flow_graph_body_impl.h

```diff
@@ -270,31 +270,68 @@ class forward_task_bypass : public graph_task {
     }
 };
 
+#if __TBB_PREVIEW_FLOW_GRAPH_TRY_PUT_AND_WAIT
+template <typename... Metainfo>
+struct graph_task_base : std::conditional<sizeof...(Metainfo) != 0,
+                                          graph_task_with_message_waiters,
+                                          graph_task>
+{};
+
+template <typename... Metainfo>
+using graph_task_base_t = typename graph_task_base<Metainfo...>::type;
+#endif
+
 //! A task that calls a node's apply_body_bypass function, passing in an input of type Input
 //  return the task* unless it is SUCCESSFULLY_ENQUEUED, in which case return nullptr
-template< typename NodeType, typename Input >
-class apply_body_task_bypass : public graph_task {
+template< typename NodeType, typename Input, typename BaseTaskType = graph_task>
+class apply_body_task_bypass
+    : public BaseTaskType
+{
     NodeType &my_node;
     Input my_input;
+
+    graph_task* call_apply_body_bypass_impl(std::true_type) {
+        return my_node.apply_body_bypass(my_input
+                                         __TBB_FLOW_GRAPH_METAINFO_ARG(message_metainfo{}));
```

Comment:
In preview mode, `apply_body_bypass` always takes a message + `message_metainfo` instance, but in regular mode - just a message. In that case, it would be no metainfo in regular mode and empty metainfo otherwise.
To be honest, this macro approach is not my invention - I just was inspired by how preview FG priorities were developed in past😄